### PR TITLE
fix a test.  Closes #1067

### LINKED ISSUE #1067: MaterializeWorkflowDescriptorActor: coercion failures
Re-enable and make the `"handle coercion failures gracefully"` test in `MaterializeWorkflowDescriptorActorSpec` work again

### PR DESCRIPTION
**Pull request by scottfrazer**

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 1]
```diff
@@ -158,10 +158,23 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef, val wor
                                       workflowOptions: WorkflowOptions,
                                       failureMode: WorkflowFailureMode,
                                       engineFileSystems: List[FileSystem]): ErrorOr[EngineWorkflowDescriptor] = {
+
+    def checkTypes(m: Map[String, WdlValue]): ErrorOr[Map[String, WdlValue]] = {
+      val allDeclarations = namespace.workflow.scopedDeclarations ++ namespace.workflow.calls.flatMap(x => x.scopedDeclarations)
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: `_.scopedDeclaration`

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 2]
```diff
@@ -158,10 +158,23 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef, val wor
...
+      val y = m map { case (k, v) =>
+        allDeclarations.find(_.fullyQualifiedName == k) match {
+          case Some(decl) if decl.wdlType.coerceRawValue(v).isFailure =>
+            s"Invalid right-side type of '$k'.  Expecting ${decl.wdlType.toWdlString}, got ${v.wdlType.toWdlString}".failureNel
+          case _ => (k, v).successNel
+        }
+      }
+      y.toList.sequence[ErrorOr, (String, WdlValue)].map(_.toMap)
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: @geoffjentry will probably disagree with me, but I'd prefer a more descriptive variable name than `y`.

Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: and maybe `m` too

**Comment by mcovarr** [comment]
:+1:

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala (hunk 2 above)
Comment (geoffjentry) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: I'd agree w/ you on `y` but not `m`, but i don't really care on either

**Comment by cjllanwarne** [comment]
👍

**Pull request merged**